src/hooks/useShiftSelection/helpers.ts

```diff
@@ -0,0 +1,262 @@
+ * @param {IOCozyFile[]} [params.items] Array of all available items (defaults to empty array)
+ * @param {SelectedItems} [params.selectedItems] Current selected items object (defaults to empty object)
+ * @param {number} params.lastInteractedIdx Index of the last interacted item
+ * @param {function} params.isItemSelected Function to check if an item is selected by _id
+ *
+ * @returns {HandleShiftSelectionResponse}
+ */
+export const handleShiftArrow = ({
+  direction,
+  items,
+  selectedItems = {},
+  lastInteractedIdx,
+  isItemSelected
+}: HandleShiftArrowParams): HandleShiftSelectionResponse => {
+  if (Object.keys(selectedItems).length === 0) {
+    const autoSelectedItem =
+      direction === FORWARD_DIRECTION ? items[0] : items[items.length - 1]
+    return {
+      newSelectedItems: {
+        [autoSelectedItem._id]: autoSelectedItem
+      },
+      lastInteractedItemId: autoSelectedItem._id
+    }
+  }
+
+  const nextIdx = lastInteractedIdx + direction
+  const currentIdx = clamp(items.length, nextIdx)
+
+  // If we hit a boundary (can't move further), keep current selection unchanged
+  if (currentIdx === lastInteractedIdx) {
+    return {
+      newSelectedItems: selectedItems,
+      lastInteractedItemId: items[lastInteractedIdx]._id
+    }
+  }
+
+  const prevSelected = isItemSelected(items[lastInteractedIdx]?._id)
+  const currSelected = isItemSelected(items[currentIdx]?._id)
+  const isMovingToSelect = prevSelected && !currSelected
+
+  const newSelectedItems = toggleSelection({
+    items,
+    selectedItems,
+    currentIdx,
+    lastInteractedIdx,
+    isMovingToSelect,
+    isItemSelected
+  })
+
+  // Updates focus position for continued navigation
+  const finalIdx = findNextBoundaryIndex({
+    items,
+    startIdx: currentIdx,
+    direction,
+    isMovingToSelect,
+    isItemSelected,
+    isReturnCurrent: Object.keys(newSelectedItems).length < 1
+  })
+
+  return {
+    newSelectedItems,
+    lastInteractedItemId: items[finalIdx]._id
+  }
+}
+
+/**
+ * Handle Shift + Click range selection.
+ * - Selects all items in range if end item is not selected
+ * - Deselects all items in range if end item is already selected
+ * - Handles reverse ranges (endIdx < startIdx) automatically
+ * - Return the last interacted item to the clicked item and new selections
+ *
+ * @param {HandleShiftClickParams} params The parameters object
+ * @param {number} params.startIdx Starting index of the selection range (last interacted item)
+ * @param {number} params.endIdx Ending index of the selection range (last clicked item)
+ * @param {SelectedItems} params.selectedItems Current selected items object
+ * @param {IOCozyFile[]} params.items Array of all available items
+ *
+ * @returns {HandleShiftSelectionResponse}
+ */
+export const handleShiftClick = ({
+  startIdx,
+  endIdx,
+  selectedItems,
+  items
+}: HandleShiftClickParams): HandleShiftSelectionResponse => {
+  let newSelectedItems = { ...selectedItems }
```

Comment:
be careful here, doing that only create a shallow copy, not a deep copy. The difference is that simple values or not related, but complex values (object and array) are. It could lead to issues in the futur. To be sure, use `cloneDeep` from lodash.

nit: Besides I'm not a big fan of `let` usage, if it's possible, maybe use reduce instead (for objects you can `Object.entries(monObjet).reduce((acc, [key, value]) => {}, something)` )